spring-cloud-contract-stub-runner/src/test/groovy/org/springframework/cloud/contract/stubrunner/AetherStubDownloaderSpec.groovy

```diff
@@ -27,4 +29,28 @@ class AetherStubDownloaderSpec extends Specification {
         then:
         jar != null
     }
+
+    def 'Should use local repository from settings.xml'() {
+        given:
+        File tempSettings = File.createTempFile("settings", ".xml")
+        final def m2repoFolder = 'm2repo/repository'
+        tempSettings.write('<settings><localRepository>' +
+                ResourceUtils.getFile(ResourceUtils.CLASSPATH_URL_PREFIX + m2repoFolder).getAbsolutePath() + '</localRepository></settings>')
+        System.setProperty("org.apache.maven.user-settings", tempSettings.getAbsolutePath())
+        final RepositorySystemSession repositorySystemSession =
```

Comment:
maybe let's add `and:` here? to separate the session creation from property settings?